pkg/workflows/builder_test.go

```diff
@@ -0,0 +1,398 @@
+package workflows_test
+
+import (
+	_ "embed"
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/yaml"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/ocr3"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/targets/chainwriter"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/triggers/streams"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/testdata/fixtures/capabilities/notstreams"
+)
+
+//go:generate go run github.com/smartcontractkit/chainlink-common/pkg/capabilities/cli/cmd/generate-types --dir $GOFILE
+
+// Note that the set of tests in this file cover the conversion from existing YAM -> code
+// along with testing the structure of what is generated from the builders.
+// This implicitly tests the code generators functionally, as the generated code is used in the tests.
+
+type Config struct {
+	Workflow    workflows.NewWorkflowParams
+	Streams     *streams.TriggerConfig
+	Ocr         *ocr3.ConsensusConfig
+	ChainWriter *chainwriter.TargetConfig
+	TargetChain string
+}
+
+func NewWorkflowSpec(rawConfig []byte) (*workflows.WorkflowSpecFactory, error) {
+	conf, err := UnmarshalYaml[Config](rawConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	workflow := workflows.NewWorkflowSpecFactory(conf.Workflow)
+	streamsTrigger := conf.Streams.New(workflow)
+	consensus := conf.Ocr.New(workflow, "ccip_feeds", ocr3.ConsensusInput{
+		Observations: workflows.ListOf[[]streams.Feed](streamsTrigger)},
+	)
+
+	conf.ChainWriter.New(workflow, conf.TargetChain, chainwriter.TargetInput{SignedReport: consensus})
+
+	return workflow, nil
+}
+
+// What if there were hundreds of feeds?  Like feeds that aren't for CCIP?
```

Comment:
Is this comment relevant to any particular field here? Maybe remove?

Reply:
It was for my demo to describe what one of the benefits of code is.  I thought leaving it shows how it can shrink the YAML and be less error-prone.  I'll remove it.